### PR TITLE
[c10d] Rename PG name and PG ID attribute

### PR DESCRIPTION
**Pull request by fduwjj**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #132915
* #132058

As discussed in https://github.com/pytorch/pytorch/pull/132058. we think pg_uid and local_uid might be a better name for pg_name and pg_id. So this PR is trying to rename it. More PRs are needed to change on the logging and other places.

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

**Review by fegin — APPROVED** [review]

File: torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp [hunk 1]
```diff
@@ -775,13 +775,13 @@ ProcessGroupNCCL::ProcessGroupNCCL(
       terminateProcessGroup_(false),
       terminateHeartbeatMonitorThread_(false),
       collectiveDebugInfoMode_(false),
-      uid_(process_group_id++),
+      local_uid_(process_group_id++),
```
Comment (wconstab) [inline torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:778]: Is local_id better? UID means unique id, it's arguably unique locally but adding the U might make people think it's more globally unique

File: torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp [hunk 2]
```diff
@@ -1593,16 +1593,17 @@ std::string ProcessGroupNCCL::createLogPrefix() const {
   if (!pg_desc_.empty() && pg_desc_ != "undefined") {
     return c10::str(
         "[PG ID ",
-        uid_,
+        local_uid_,
         "PG GUID ",
-        pg_name_,
+        pg_uid_,
```
Comment (wconstab) [inline torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:1598]: Is pg_global_uid better? I'm ok with uid too if we take u out of local 

Unlike local id which is a simple counter, this variable has special properties. I want to have a hint about that from the name.

Comment (fduwjj) [reply torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:1598]: I prefer taking out u from local. adding global here seems a little bit duplicate.

**Pull request closed without merging**